app/code/Magento/InventoryCatalog/Plugin/Model/StockRepositoryPlugin.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\InventoryCatalog\Plugin\Model;
+
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\InventoryApi\Api\StockRepositoryInterface;
+use Magento\InventoryCatalog\Api\DefaultStockProviderInterface;
+
+/**
+ * Plugin Stock Repository
+ */
+class StockRepositoryPlugin
+{
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
+    /**
+     * @param DefaultStockProviderInterface $defaultStockProvider
+     */
+    public function __construct(DefaultStockProviderInterface $defaultStockProvider)
+    {
+        $this->defaultStockProvider = $defaultStockProvider;
+    }
+
+    /**
+     * Prevent default source to be deleted
+     *
+     * @param StockRepositoryInterface $subject
+     * @param int $stockId
+     *
+     * @return null
```

Comment:
return void

Reply:
it is from http://devdocs.magento.com/guides/v2.2/extension-dev-guide/plugins.html "If the method does not change the argument for the observed method, it should return null" but maybe my thinking was wrong in this case.